### DISCUSSION TITLE
How to use materials from assimp?

### DISCUSSION THREAD

**Discussion by cj2000** [assimp/assimp#4559] [Q&A]
After I have imported a model with assimp, I would like to visualise it in OpenGL. While the geometry is no problem to draw, have no idea how to aply the materials.

**Comment by kimkulling** [comment]
Check this set of tutorials: https://learnopengl.com/Model-Loading/Assimp 

Here you will find a couple of great tutorials how to render textures.

Kim

**Reply by cj2000** [reply]
Looks like misunderstanding, so I beter write in german, as this is the language I can beter explain:

Es geht eigentlich darum wie ich aus dem aiMaterial solche Sachen wie difuse, ambiente und specular color so wie die Texturnamen rauskriege. Wenn ich alle diese Werte habe ist es unproblematisch für mich diese in OpenGL zu verwenden.

**Comment by kimkulling** [comment]
That is easy. Just check the samples like: [SimpleOpenGL.c](https://github.com/assimp/assimp/blob/master/samples/SimpleOpenGL/Sample_SimpleOpenGL.c)

For instance: 

```
void apply_material(const C_STRUCT aiMaterial *mtl)
{
	float c[4];

	GLenum fill_mode;
	int ret1, ret2;
	C_STRUCT aiColor4D diffuse;
	C_STRUCT aiColor4D specular;
	C_STRUCT aiColor4D ambient;
	C_STRUCT aiColor4D emission;
	ai_real shininess, strength;
	int two_sided;
	int wireframe;
	unsigned int max;

	set_float4(c, 0.8f, 0.8f, 0.8f, 1.0f);
	if(AI_SUCCESS == aiGetMaterialColor(mtl, AI_MATKEY_COLOR_DIFFUSE, &diffuse))
		color4_to_float4(&diffuse, c);
	glMaterialfv(GL_FRONT_AND_BACK, GL_DIFFUSE, c);

	set_float4(c, 0.0f, 0.0f, 0.0f, 1.0f);
	if(AI_SUCCESS == aiGetMaterialColor(mtl, AI_MATKEY_COLOR_SPECULAR, &specular))
		color4_to_float4(&specular, c);
	glMaterialfv(GL_FRONT_AND_BACK, GL_SPECULAR, c);

	set_float4(c, 0.2f, 0.2f, 0.2f, 1.0f);
	if(AI_SUCCESS == aiGetMaterialColor(mtl, AI_MATKEY_COLOR_AMBIENT, &ambient))
		color4_to_float4(&ambient, c);
	glMaterialfv(GL_FRONT_AND_BACK, GL_AMBIENT, c);
```

**Reply by cj2000** [reply]
And what about textures of the material?
